django/contrib/staticfiles/management/commands/collectstatic.py

```diff
@@ -279,7 +279,7 @@ def delete_file(self, path, prefixed_path, source_storage):
                     # Skip the file if the source file is younger
                     # Avoid sub-second precision (see #14665, #19540)
                     if (target_last_modified.replace(microsecond=0) >= source_last_modified.replace(microsecond=0) and
-                            full_path and not (self.symlink ^ os.path.islink(full_path))):
+                            not full_path or full_path and not (self.symlink ^ os.path.islink(full_path))):
```

Comment:
Having `not full_path or full_path` in the middle of this expression is extremely confusing.

Could you add parentheses instead of relying on operator precedence?

`(a and b) or (c and d)` takes significantly less effort to parse than `a and b or c and d`.